synapse/cortex.py

```diff
@@ -6391,6 +6393,76 @@ async def _setVaultSecrets(self, iden, obj, key, valu, delete):
         self.slab.put(bidn, s_msgpack.en(vault), db=self.vaultsdb)
         return data
 
+    async def replaceVaultConfigs(self, iden, valu):
+        '''
+        Replace the entire vault config.
+
+        Args:
+            iden (str): The iden of the vault to edit.
+            valu (str): New configs object to store on the vault.
+
+        Raises:
+            synapse.exc.BadArg: `valu` is not a dictionary.
+            synapse.exc.NoSuchIden: Vault with `iden` does not exist.
+            synapse.exc.NotMsgpackSafe: `valu` is not msgpack safe.
+
+        Returns: New configs.
+        '''
+        vault = self.reqVault(iden)
+
+        if not isinstance(valu, dict):
+            raise s_exc.BadArg(mesg='valu must be a dictionary.', name='replaceVaultConfigs', valu=valu)
+
+        try:
+            s_msgpack.en(valu)
+        except s_exc.NotMsgpackSafe:
+            raise s_exc.NotMsgpackSafe(
+                mesg='Vault configs must be msgpack safe.',
+                name='replaceVaultSecrets',
```

Comment:
Copy pasta?  Also, I think the use of name of name here (and in BadArg above) would typically be the arg name not the method.

Reply:
Yep, good catch. I'll update